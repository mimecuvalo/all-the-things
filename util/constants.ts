const constants = {
  // Add your shared client/server constants here.
};

export default constants;
