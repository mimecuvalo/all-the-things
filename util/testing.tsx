import React, { FC, ReactElement } from 'react';
import { RenderOptions, render } from '@testing-library/react';

import { ApolloProvider } from '@apollo/client';
import { IntlProvider } from 'i18n';
import { ThemeProvider } from '@mui/material/styles';
import createApolloClient from 'app/apollo';
import { muiTheme } from 'styles';

const AllTheProviders: FC = ({ children }) => {
  const client = createApolloClient();

  return (
    <ThemeProvider theme={muiTheme}>
      <ApolloProvider client={client}>
        <IntlProvider defaultLocale="en" locale="en" messages={{}}>
          {children}
        </IntlProvider>
      </ApolloProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
