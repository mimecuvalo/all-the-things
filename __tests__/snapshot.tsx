import Home from '@/pages';
import { render } from 'util/testing';

it('renders homepage unchanged', () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
