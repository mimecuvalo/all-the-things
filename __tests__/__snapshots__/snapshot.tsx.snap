// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main notranslate"
    >
      <div
        style="position: relative; opacity: 0; bottom: 100px;"
      >
        <img
          alt="logo"
          class="appLogo"
          data-nimg="1"
          decoding="async"
          height="72"
          loading="lazy"
          src="/_next/image?url=%2Ffavicon.jpg&w=256&q=75"
          srcset="/_next/image?url=%2Ffavicon.jpg&w=96&q=75 1x, /_next/image?url=%2Ffavicon.jpg&w=256&q=75 2x"
          style="color: transparent;"
          width="72"
        />
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h1 title css-lgx1dj-MuiTypography-root"
      >
        Welcome to 
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-18sjyu7-MuiTypography-root-MuiLink-root"
          href="https://nextjs.org"
        >
          Next.js
        </a>
        , with
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-18sjyu7-MuiTypography-root-MuiLink-root"
          href="https://github.com/mimecuvalo/all-the-things"
        >
          All The Things
        </a>
        !
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code
          class="code"
        >
          pages/index.tsx
        </code>
      </p>
      <div
        class="grid"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover card css-18sjyu7-MuiTypography-root-MuiLink-root"
          href="https://nextjs.org/docs"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1icjozd-MuiTypography-root"
          >
            Documentation →
          </h2>
          <p>
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover card css-18sjyu7-MuiTypography-root-MuiLink-root"
          href="https://nextjs.org/learn"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1icjozd-MuiTypography-root"
          >
            Learn →
          </h2>
          <p>
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover card css-18sjyu7-MuiTypography-root-MuiLink-root"
          href="https://github.com/vercel/next.js/tree/canary/examples"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1icjozd-MuiTypography-root"
          >
            Examples →
          </h2>
          <p>
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover card css-18sjyu7-MuiTypography-root-MuiLink-root"
          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1icjozd-MuiTypography-root"
          >
            Deploy →
          </h2>
          <p>
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1icjozd-MuiTypography-root"
      >
        <span
          class="i18n-msg"
        >
          All The Things Feature test
        </span>
      </h2>
      <p
        class="p"
      >
        <span
          class="i18n-msg"
        >
          GraphQL variables test (current url path): 
        </span>
      </p>
      <p
        class="p"
      >
        <span
          class="i18n-msg"
        >
          Experiment disabled
        </span>
      </p>
      <p
        class="p"
      >
        <span
          class="i18n-msg"
        >
          i18n pluralization test: 5,000 items.
        </span>
      </p>
      <p
        class="p"
      >
        <span
          class="i18n-msg"
        >
          i18n html test: 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover external-link css-18sjyu7-MuiTypography-root-MuiLink-root"
            href="https://www.example.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            visit our website
          </a>
           and 
          <strong>
            see the world
          </strong>
        </span>
      </p>
      <p>
        and 
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-18sjyu7-MuiTypography-root-MuiLink-root"
          href="https://github.com/mimecuvalo/all-the-things"
        >
          much more
        </a>
        !
      </p>
    </main>
    <footer
      class="footer"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover notranslate css-18sjyu7-MuiTypography-root-MuiLink-root"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by
         
        <span
          class="logo"
        >
          <img
            alt="Vercel Logo"
            data-nimg="1"
            decoding="async"
            height="16"
            loading="lazy"
            src="/img/vercel.svg"
            srcset="/img/vercel.svg 1x, /img/vercel.svg 2x"
            style="color: transparent;"
            width="72"
          />
        </span>
      </a>
    </footer>
  </div>
</div>
`;
